### ISSUE TITLE
Fixing import cudf in a Jupyter notebook [FEA]

### ISSUE THREAD

**Issue by astrophysaxist** [rapidsai/cudf#8128]
Hi, I've just started playing around with RAPIDS in a virtual environment I setup on a machine I don't have root access to, and it's been a pain troubleshooting how to get cuDF to import without an error in the libstdc++ library.  It'd be nice if there was a way to import cudf in a Jupyter notebook that doesn't involve copying the environment libraries to root directories.

After installing RAPIDS in a new environment using: `conda create -n rapids-0.19 -c rapidsai -c nvidia -c conda-forge rapids=0.19  python=3.7 cudatoolkit=11.2`

Opening a `jupyter notebook` and typing `import cudf` yields a Traceback that ends in:
```
ImportError: /software/linux/usr1/local/anaconda_py3/anaconda3-2018.12-Linux-x86_64/lib/python3.7/site-packages/zmq/backend/cython/../../../../.././libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /local/data/home/[user]/.conda/envs/rapids-0.19/lib/python3.7/site-packages/cudf/_lib/../../../../libarrow.so.100)
```

It looks like this is sort of a known issue since apparently it's a problem for BlazingSQL in one of the sample RAPIDS Google Colab [notebooks](https://colab.research.google.com/drive/1rY7Ln6rEE1pOlfSHCYOVaqt8OvDO35J0#forceEdit=true&offline=true&sandboxMode=true). Relevant fix there is:
```
# fix for BlazingSQL import issue
# ImportError: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /usr/local/lib/python3.7/site-packages/../../libblazingsql-engine.so)
if not os.path.exists('/usr/lib64'):
    os.makedirs('/usr/lib64')
for so_file in os.listdir('/usr/local/lib'):
  if 'libstdc' in so_file:
    shutil.copyfile('/usr/local/lib/'+so_file, '/usr/lib64/'+so_file)
    shutil.copyfile('/usr/local/lib/'+so_file, '/usr/lib/x86_64-linux-gnu/'+so_file)
```

But, I can't implement that fix in my current setup because I don't have permissions on that directory. I'm not sure if others have found a fix besides renaming and copying libraries, but I haven't seen any.

**Comment by kkraus14** [comment]
@astrophysaxist this sounds like there's a different `libstdc++.so.6` set in `LD_LIBRARY_PATH` that's too old of a version.

Could you dump the contents of a `printenv` here for debugging?

**Comment by astrophysaxist** [comment]
Hi @kkraus14 , sure see below:

```
PROJ_LIB=/local/data/home/[USER]/.conda/envs/rapids-0.19/share/proj
CUDA_PATH=/local/data/home/[USER]/.conda/envs/rapids-0.19
MANPATH=/usr/share/man:/usr/man:/usr1/local/man:/usr/X11R6/man
XDG_SESSION_ID=2473
HOSTNAME=[HOSTNAME]
SELINUX_ROLE_REQUESTED=
TERM=xterm-256color
SHELL=/bin/bash
HISTSIZE=1000
TMPDIR=/local/data/[machine]/[USER]/tmp
CONDA_SHLVL=1
CONDA_PROMPT_MODIFIER=(rapids-0.19) 
SELINUX_USE_CURRENT_RANGE=
QTDIR=/usr/lib64/qt-3.3
OLDPWD=/local/data/home/[USER]
QTINC=/usr/lib64/qt-3.3/include
SSH_TTY=/dev/pts/1
QT_GRAPHICSSYSTEM_CHECKED=1
USER=[USER]
LD_LIBRARY_PATH=/usr/lib:/usr/lib/X11:/usr1/local/lib
LS_COLORS=rs=0:di=38;5;27:ln=38;5;51:mh=44;38;5;15:pi=40;38;5;11:so=38;5;13:do=38;5;5:bd=48;5;232;38;5;11:cd=48;5;232;38;5;3:or=48;5;232;38;5;9:mi=05;48;5;232;38;5;15:su=48;5;196;38;5;15:sg=48;5;11;38;5;16:ca=48;5;196;38;5;226:tw=48;5;10;38;5;16:ow=48;5;10;38;5;21:st=48;5;21;38;5;15:ex=38;5;34:*.tar=38;5;9:*.tgz=38;5;9:*.arc=38;5;9:*.arj=38;5;9:*.taz=38;5;9:*.lha=38;5;9:*.lz4=38;5;9:*.lzh=38;5;9:*.lzma=38;5;9:*.tlz=38;5;9:*.txz=38;5;9:*.tzo=38;5;9:*.t7z=38;5;9:*.zip=38;5;9:*.z=38;5;9:*.Z=38;5;9:*.dz=38;5;9:*.gz=38;5;9:*.lrz=38;5;9:*.lz=38;5;9:*.lzo=38;5;9:*.xz=38;5;9:*.bz2=38;5;9:*.bz=38;5;9:*.tbz=38;5;9:*.tbz2=38;5;9:*.tz=38;5;9:*.deb=38;5;9:*.rpm=38;5;9:*.jar=38;5;9:*.war=38;5;9:*.ear=38;5;9:*.sar=38;5;9:*.rar=38;5;9:*.alz=38;5;9:*.ace=38;5;9:*.zoo=38;5;9:*.cpio=38;5;9:*.7z=38;5;9:*.rz=38;5;9:*.cab=38;5;9:*.jpg=38;5;13:*.jpeg=38;5;13:*.gif=38;5;13:*.bmp=38;5;13:*.pbm=38;5;13:*.pgm=38;5;13:*.ppm=38;5;13:*.tga=38;5;13:*.xbm=38;5;13:*.xpm=38;5;13:*.tif=38;5;13:*.tiff=38;5;13:*.png=38;5;13:*.svg=38;5;13:*.svgz=38;5;13:*.mng=38;5;13:*.pcx=38;5;13:*.mov=38;5;13:*.mpg=38;5;13:*.mpeg=38;5;13:*.m2v=38;5;13:*.mkv=38;5;13:*.webm=38;5;13:*.ogm=38;5;13:*.mp4=38;5;13:*.m4v=38;5;13:*.mp4v=38;5;13:*.vob=38;5;13:*.qt=38;5;13:*.nuv=38;5;13:*.wmv=38;5;13:*.asf=38;5;13:*.rm=38;5;13:*.rmvb=38;5;13:*.flc=38;5;13:*.avi=38;5;13:*.fli=38;5;13:*.flv=38;5;13:*.gl=38;5;13:*.dl=38;5;13:*.xcf=38;5;13:*.xwd=38;5;13:*.yuv=38;5;13:*.cgm=38;5;13:*.emf=38;5;13:*.axv=38;5;13:*.anx=38;5;13:*.ogv=38;5;13:*.ogx=38;5;13:*.aac=38;5;45:*.au=38;5;45:*.flac=38;5;45:*.mid=38;5;45:*.midi=38;5;45:*.mka=38;5;45:*.mp3=38;5;45:*.mpc=38;5;45:*.ogg=38;5;45:*.ra=38;5;45:*.wav=38;5;45:*.axa=38;5;45:*.oga=38;5;45:*.spx=38;5;45:*.xspf=38;5;45:
CONDA_EXE=/usr1/local/anaconda_py3/anaconda3-2018.12-Linux-x86_64/bin/conda
_CE_CONDA=
CPL_ZIP_ENCODING=UTF-8
MAIL=/var/spool/mail/[USER]
PATH=/local/data/home/[USER]/.conda/envs/rapids-0.19/bin:/usr1/local/anaconda_py3/anaconda3-2018.12-Linux-x86_64/condabin:/usr1/local/anaconda_py3/anaconda3-2018.12-Linux-x86_64/bin:/usr1/local/anaconda/bin:/usr/ucb:/usr/bin/X11:/usr1/local/gnu/bin:/usr/etc:/usr/bin:/usr1/local/anaconda/bin:.:/usr/local/bin:/usr1/local/bin:/bin:/local/data/home/[USER]/bin:.:/local/data/home/[USER]/.local/bin
PROJ_NETWORK=ON
CONDA_PREFIX=/local/data/home/[USER]/.conda/envs/rapids-0.19
PWD=/local/data/[machine]/[USER]
LANG=en_US.utf-8
KDEDIRS=/usr
XFILESEARCHPATH=/usr/lib/X11/%T/%N:/usr1/local/lib/%T/%N:/usr/share/X11/%T/%N
PS1=(rapids-0.19) \[\033[0;36m\]\h\[\033[0m\]:\W \[\033[0;32m\]\u\[\033[0m\]\!\$ 
SELINUX_LEVEL_REQUESTED=
GDAL_DATA=/local/data/home/[USER]/.conda/envs/rapids-0.19/share/gdal
_CE_M=
HISTCONTROL=ignoredups
GNUBINDIR=/usr1/local/gnu/bin
SHLVL=1
HOME=/local/data/home/[USER]
CONDA_PYTHON_EXE=/usr1/local/anaconda_py3/anaconda3-2018.12-Linux-x86_64/bin/python
LESS=-M
PYTHONPATH=:/local/data/home/[USER]
LOGNAME=[USER]
PRINTER=xerox-b34-rs259
QTLIB=/usr/lib64/qt-3.3/lib
XDG_DATA_DIRS=/local/data/home/[USER]/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share
CONDA_DEFAULT_ENV=rapids-0.19
LESSOPEN=||/usr/bin/lesspipe.sh %s
XDG_RUNTIME_DIR=/run/user/2423
QT_PLUGIN_PATH=/usr/lib64/kde4/plugins:/usr/lib/kde4/plugins
_=/usr/bin/printenv
```
I'm starting to think there must be something unexpected in how calling my `jupyter notebook` command in the environment affects which libraries are loaded.

**Comment by kkraus14** [comment]
@astrophysaxist agreed. Nothing looks off in the above environment variables. Do you have `jupyter` / `ipython` installed in your environment? What does `which jupyter` and `which ipython` return?

**Comment by astrophysaxist** [comment]
Interestingly enough, the environment has jupyter and ipython coming from the right location:

```
(rapids-0.19) :$ which jupyter
~/.conda/envs/rapids-0.19/bin/jupyter
(rapids-0.19) :$ which ipython
~/.conda/envs/rapids-0.19/bin/ipython
```
When I've been starting up a remote jupyter notebook session though I do see messages that lead me to suspect it's still grabbing the machine's jupyter and note the environments. It seems to grab the rapids-0.19 JupyterLab application but uses the machine's JupyterLab extension.

**Comment by astrophysaxist** [comment]
Small update here, I finally chased this down to the system `zmq` install that was being picked up in my jupyter session. I was able install Python 3.8 in an environment and a new version of `pyzmq` and the import of `cudf` worked. Still have to go back and trace my steps to verify the system/global `zmq` being picked up caused this.  I'm guessing that by running the default conda install of rapids at the Getting Started page the system zmq libraries must have gotten found, and since they point to older libraries the import failed. I'll try to replicate and come up with the simplest set of install steps.
